rust/core-lib/src/event_context.rs

```diff
@@ -211,14 +211,20 @@ impl<'a> EventContext<'a> {
         let delta = if approx_size > MAX_SIZE_LIMIT { None } else { Some(delta) };
 
         let undo_group = ed.get_active_undo_group();
+        //TODO: we want to just put EditType on the wire, but don't want
+        //to update the plugin lib quite yet.
+        let edit_type_str = serde_json::to_string(&ed.get_edit_type())
```

Comment:
This is janky, but I understand it won't be the final state of the code. Would it be possible to serde to a json::Value instead of all the way to a string?